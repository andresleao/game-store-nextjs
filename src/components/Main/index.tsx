import * as S from './styles';

const Main = ({
  title = 'React Avançado',
  description = 'Typescript ReactJS, NextJS e Styled Components'
}) => {
  return (
    <S.Wrapper>
      <S.Logo
        src="/img/logo.svg"
        alt="Imagem de um átomo"
      />
      <S.Title>{title}</S.Title>
      <S.Description>{description}</S.Description>
      <S.Illustration
        src="/img/hero-illustration.svg"
        alt="Imagem de um desenvolvedor de frente para a tela de um computador"
      />
      <h1>React Avançado</h1>
    </S.Wrapper>
  );
};

export default Main;
