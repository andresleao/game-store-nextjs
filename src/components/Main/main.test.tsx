import Main from '.';
import { renderWithTheme } from '@/utils/tests/helpers';

describe('<Main />', () => {
  it('Deve renderizar o heading', () => {
    // Renderiza o componente
    //const { container, getByRole } = render(<Main />);
    const { container, getAllByRole } = renderWithTheme(<Main />);

    // Busca o elemento e verifica a existência
    expect(getAllByRole('heading', { name: /react avançado/i })[0]).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Deve renderizar a cor corretamente', () => {
    const { container } = renderWithTheme(<Main />);

    expect(container.firstChild).toHaveStyle({'background-color': '#06092b'});
  });
});
