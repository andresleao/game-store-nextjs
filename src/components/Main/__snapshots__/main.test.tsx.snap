// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Deve renderizar o heading 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.c1 {
  width: 25rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 2rem;
  font-weight: 400;
}

.c4 {
  margin-top: 3rem;
  width: min(30rem, 100%);
}

<main
  class="c0"
>
  <img
    alt="Imagem de um átomo"
    class="c1"
    src="/img/logo.svg"
  />
  <h1
    class="c2"
  >
    React Avançado
  </h1>
  <h2
    class="c3"
  >
    Typescript ReactJS, NextJS e Styled Components
  </h2>
  <img
    alt="Imagem de um desenvolvedor de frente para a tela de um computador"
    class="c4"
    src="/img/hero-illustration.svg"
  />
  <h1>
    React Avançado
  </h1>
</main>
`;
